graylog2-web-interface/src/components/bootstrap/BootstrapModalForm.jsx

```diff
@@ -18,12 +18,12 @@ class BootstrapModalForm extends Component {
     this.open = this.open.bind(this);
     this.close = this.close.bind(this);
     this._submit = this._submit.bind(this);
-    this._onModalClose = this._onModalClose.bind(this);
+    this._onModalCancel = this._onModalCancel.bind(this);
   }
 
-  _onModalClose() {
-    if (typeof this.props.onModalClose === 'function') {
-      this.props.onModalClose();
+  _onModalCancel() {
+    if (typeof this.props.onCancel === 'function') {
```

Comment:
I still don't see why we need to differentiate both cases. I think submitting the form also needs to reset the inputs in some cases (after creating an entity, for instance). I also think that some people will be mad because they pressed cancel instead of the "x" or escape and their inputs are gone, specially in forms using textareas like an alert callback email template or a sidecar snippet. At least I would be mad if I needed to re-enter the input for that reason. At the moment we are fairly inconsistent with that (as with many other things), but I see this as a step backwards in usability.

Regarding `onModalClose` not being called, I meant on external components using `BootstrapModalForm`. See [this example](https://github.com/Graylog2/graylog2-server/blob/c8e26cb9602bc4ceb95c4b77469dabb8f71b3f09/graylog2-web-interface/src/components/configurations/MessageProcessorsConfig.jsx#L157-L157) for instance: The `MesageProcessorConfig` expects the `onModalClose` callback to be called on close or cancel, but now it will only be called on close, changing the previous behaviour of that component.